### PR TITLE
Linter

### PR DESCRIPTION
**Pull request by maximzubkov**
Add cpplint to workflow

### CODE REVIEW

File: scripts/install_requirements.sh [hunk 1]
```diff
@@ -1,7 +1,8 @@
 #!/bin/bash
-# Run script from code2seq dir using command:
+# Run script from gorshochek dir using command:
 #    sh scripts/install_requirements.sh
 
+pip install -r requirements.txt
```
Comment (SpirinEgor) [inline scripts/install_requirements.sh]: `cpplint` doesn't require to run this project. And, as I understand, we need to install clang for making transformations and we need to install some python packages for development. There aren't any other requirements. So, let's rename this script to `install_clang.sh` and call `pip install` manually.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -10,12 +10,17 @@ jobs:
...
     steps:
       - uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
       - name: Install dependencies
         run: |
+          python -m pip install --upgrade pip
           sh scripts/install_requirements.sh
```
Comment (SpirinEgor) [inline .github/workflows/ci.yml]: `pip install -r requirements.txt`

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -10,12 +10,17 @@ jobs:
...
       - name: Lint
         run: |
-          echo "Linting is not implemented yet"
+          cpplint --filter=-legal,-build main.cpp
```
Comment (SpirinEgor) [inline .github/workflows/ci.yml]: So, we need to specify all files here? Is it possible to pass only the root folder?

Comment (SpirinEgor) [reply .github/workflows/ci.yml]: Btw, what do these filters do?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (SpirinEgor) [reply .github/workflows/ci.yml]: And we need one more step for install clang

File: README.md [hunk 4]
```diff
@@ -8,7 +8,7 @@ To install `clang+llvm-9` to use only inside `gorshochek` following commands sho
 git clone https://github.com/JetBrains-Research/gorshochek.git
 cd gorshochek
 
-# Install clang+llvm-9
+# Install clang+llvm-9, cpplint
```
Comment (SpirinEgor) [inline README.md]: Update usage instructions to this:

## Usage
1. Clone or download repo from GitHub
```(bash)
git clone https://github.com/JetBrains-Research/gorshochek.git
cd gorshochek
```
2. Install clang with the desired version to use only inside `gorshochek`
```(bash)
./scripts/install_clang.sh
```
3. Build the project from source
```(bash)

cmake -DLLVM_DIR=build/clang+llvm-9/lib/cmake/llvm \
      -DClang_DIR=build/clang+llvm-9/lib/cmake/clang \
      -DCMAKE_CXX_COMPILER=/usr/bin/c++ ..
cmake --build .
```
4. Run the tool with the needed configuration on C++ files
```(bash)
# TODO
```

**Review by SpirinEgor — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -10,12 +10,18 @@ jobs:
...
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
       - name: Install dependencies
         run: |
-          sh scripts/install_requirements.sh
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          sh scripts/install_clang.sh
```
Comment (SpirinEgor) [inline .github/workflows/ci.yml:23]: Move it to the next step.
One step for install python dependencies, another for install clang.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (maximzubkov) [reply .github/workflows/ci.yml]: Filter `legal` targets copyright issues, the second one `build` I decided to remove

File: README.md [hunk 6]
```diff
@@ -2,19 +2,27 @@
 [WIP] A tool for C++ code modification to augment data for clone detection tools
 
 -------------
-### Install requirements and build
-To install `clang+llvm-9` to use only inside `gorshochek` following commands should be executed: 
-```
+Update usage instructions to this:
+
```
Comment (SpirinEgor) [inline README.md]: Lol, I don't mean to copy the full message into readme.

File: README.md [hunk 7]
```diff
@@ -2,19 +2,27 @@
...
+./scripts/install_clang.sh
+```
+3. Build the project from source
+```(bash)
 
-# Install clang+llvm-9
-sh scripts/install_requirements.sh
-
-# Build
-cd build
```
Comment (SpirinEgor) [inline README.md:15]: I think we should keep this line. It seems I forgot about it.

**Review by SpirinEgor — CHANGES_REQUESTED** [review]

**Review by SpirinEgor — APPROVED** [review]

**Pull request merged**